styx-common/src/main/java/com/spotify/styx/storage/TransactionalStorage.java

```diff
@@ -0,0 +1,82 @@
+/*-
+ * -\-\-
+ * Spotify Styx Common
+ * --
+ * Copyright (C) 2016 - 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.storage;
+
+import com.spotify.styx.model.Workflow;
+import com.spotify.styx.model.WorkflowId;
+import com.spotify.styx.model.WorkflowState;
+import com.spotify.styx.util.TriggerInstantSpec;
+import java.io.IOException;
+
+/**
+ * The interface to the persistence layer where the same transaction can be used across storage
```

Comment:
The use of "can" here... I think the design we have is "the same transaction _is_ used across storage operations", should we document that clearly?

Reply:
The naming choice also made me initially think that a "TransactionalStorage" would manage several transactions. If not, then perhaps let's not hide that it _is a_ (wrapper of a) transaction? `com.spotify.styx.storage.Transaction`? `com.spotify.styx.storage.StorageTransaction`?

See how in `TransactionFunction.apply` the parameter is called `tx`, short for "transaction", yet the type name contradicts that reading, it doesn't read as "FooBarTransaction" but rather `TransactionalStorage`...

Furthermore in `DatastoreStorage` we have `transactionalStorageFactory`, suggesting that we produce different storages out of given storage. At this point I as a reader am confused what such stacking of Storage abstractions means, and it's probably even harder to grasp for new readers... Actually, in common usage terms, the objects that the factory produces are wrappers of _single transactions_ against the same `DatastoreStorage`, so, huh...